.github/workflows/continuous-integration.yml

```diff
@@ -393,7 +457,18 @@ jobs:
       - name: Create test results directory
         run: mkdir -p test-results
 
-      - name: Run Cypress tests
+      - name: Run Cypress tests on push
+        if: ${{ github.ref != 'refs/heads/master' }}
+        run: node script/github-actions/run-selected-cypress-tests.js
+        timeout-minutes: 30
+        env:
+          CYPRESS_CI: true
+          STEP: ${{ matrix.ci_node_index }}
+          NUM_CONTAINERS: ${{ needs.cypress-tests-prep.outputs.num_containers }}
+          TESTS: ${{ needs.cypress-tests-prep.outputs.tests }}
+
+      - name: Run Cypress tests on merge to master
+        if: ${{ github.ref == 'refs/heads/master' }}
         run: node script/github-actions/run-cypress-tests.js
         timeout-minutes: 30
         env:
```

Comment:
Is the `NUM_CONTAINERS` variable needed here?

Reply:
No, it's not. That step runs the `script/github-actions/run-cypress-tests.js` file which hasn't changed.